frontends/web/src/components/aopp/aopp.tsx

```diff
@@ -0,0 +1,127 @@
+    };
+
+    private chooseAccount = (code: AccountCode) => {
+        backendAPI.aoppChooseAccount(code);
+    }
+
+    public render(
+        { t, aopp }: RenderableProps<Props>,
+    ) {
+        if (!aopp) {
+            return null;
+        }
+        switch (aopp.state) {
+            case 'error':
+                return (
+                    <div>
+                        <p>{ t('aopp.addressRequested', { host: aopp.callbackHost }) }</p>
+                        <p>{ t(`error.${aopp.errorCode}`, { host: aopp.callbackHost }) }</p>
+                        <Button primary onclick={backendAPI.aoppCancel}>Dismiss</Button>
+                    </div>
+                );
+            case 'inactive':
+                // Inactive, waiting for action.
+                return null;
+            case 'awaiting-keystore':
+                return (
+                    <div>
+                        <p>{ t('aopp.addressRequested', { host: aopp.callbackHost }) }</p>
+                        <p>{ t('aopp.awaitingKeystore') }</p>
+                        <Button primary onclick={backendAPI.aoppCancel}>Cancel</Button>
+                    </div>
+                );
+            case 'choosing-account': {
+                const options = aopp.accounts.map(account => {
+                    return {
+                        text: account.name,
+                        value: account.code,
+                    };
+                });
+                return (
+                    <div>
+                        <Select
+                            options={[{
+                                text: t('buy.info.selectLabel'),
+                                disabled: true,
+                                value: 'choose',
+                            }, ...options]
+                            }
+                            defaultValue={'choose'}
+                            onChange={e => this.chooseAccount(e.target.value)}
+                            id="account"
+                        />
+                        <Button primary onclick={backendAPI.aoppCancel}>Cancel</Button>
+                    </div>
+                );
+            }
+            case 'syncing':
+                return <div>Syncing the account, please stand by.</div>;
+            case 'signing':
+                return (
+                    <div>
+                        <p>Address: {aopp.address}</p>
+                        <p>Please confirm on your BitBox.</p>
+                    </div>
+                );
+            case 'success':
+                return (
+                    <div>
+                        <p>Successfully delivered a fresh address to the third party.</p>
+                        <Button primary onclick={backendAPI.aoppCancel}>Dismiss</Button>
+                    </div>
+                );
+        }
+    }
+}
+
+const subscribeHOC = subscribe<SubscribedProps, AoppProps & TranslateProps>(
+    { aopp: 'aopp' },
+    false,
+    false,
+)(Aopp);
```

Comment:
I have really a hard time doing this, because in my mind, the subscribe decorator is perfect. It does the apiGet on first load, it gets the subsequent notifications, it does not clutter the component with multiple calls to do the same in didMount... what is not to like about it?

Reply:
lets save this discussing for another day.